### PR TITLE
Lambda sink refactor

### PR DESCRIPTION
**Pull request by srikanthjg**
### Description
These changes refactor existing lambda sink to accommodate lambda processor changes.
 
### Issues Resolved
Contribues to  #4699 
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/aws-lambda/build.gradle [hunk 1]
```diff
@@ -59,9 +61,13 @@ task integrationTest(type: Test) {
...
 
     systemProperty 'log4j.configurationFile', 'src/test/resources/log4j2.properties'
-    systemProperty 'tests.sink.lambda.region', System.getProperty('tests.sink.lambda.region')
-    systemProperty 'tests.sink.lambda.functionName', System.getProperty('tests.sink.lambda.functionName')
-    systemProperty 'tests.sink.lambda.sts_role_arn', System.getProperty('tests.sink.lambda.sts_role_arn')
+    systemProperty 'tests.lambda.sink.region', System.getProperty('tests.lambda.sink.region')
+    systemProperty 'tests.lambda.sink.functionName', System.getProperty('tests.lambda.sink.functionName')
+    systemProperty 'tests.lambda.sink.sts_role_arn', System.getProperty('tests.lambda.sink.sts_role_arn')
+
+    systemProperty 'tests.lambda.processor.region', System.getProperty('tests.lambda.processor.region')
```
Comment (kkondaka) [inline data-prepper-plugins/aws-lambda/build.gradle:68]: remove processor properies?

File: data-prepper-plugins/aws-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+
+# Lambda Processor
```
Comment (kkondaka) [inline data-prepper-plugins/aws-lambda/README.md:2]: remove processor doc?

Comment (oeyh) [reply data-prepper-plugins/aws-lambda/README.md:2]: Let's keep the developer guide for running integ tests here. We can remove the usage sections after they are added to Opensearch documentation website.

Comment (srikanthjg) [reply data-prepper-plugins/aws-lambda/README.md:2]: do we want to have 2 different readme for lambda processor and sink?

**Review by kkondaka — APPROVED** [review]

**Review by oeyh — APPROVED** [review]

**Pull request merged**